frontend/src/metabase/visualizations/echarts/cartesian/chart-measurements/index.ts

```diff
@@ -102,13 +108,18 @@ const getXAxisTicksWidth = (
     family: fontFamily,
   };
 
-  const firstXTickWidth = measureText(
-    formatter(dataset[0][X_AXIS_DATA_KEY]),
-    fontStyle,
-  );
-  const lastXTickWidth = measureText(
-    formatter(dataset[dataset.length - 1][X_AXIS_DATA_KEY]),
-    fontStyle,
+  const valuesToMeasure = [0, dataset.length - 1].map(index => {
```

Comment:
xAxisModel has `extent`, can we use it here?

Reply:
Doesn't exist for categorical or time series axes. We could do separate logic for value and other axes but it would be more verbose